# Prevents libpng's build system from using AWK.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -351,7 +351,7 @@ endif()
 # Find an AWK language processor.
 # Start with specific AWK implementations like gawk and nawk, which are
 # known to work with our scripts, then fall back to the system awk.
-find_program(AWK NAMES gawk nawk awk)
+
 if(AWK)
   message(STATUS "Found AWK program: ${AWK}")
 else()
