# Makes pixman's build system inherit our compiler flags.

--- a/meson.options
+++ b/meson.options
@@ -18,6 +18,9 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+option('cflags', type: 'string')
+option('cppflags', type: 'string')
+
 option(
   'loongson-mmi',
   type : 'feature',
--- a/pixman/meson.build
+++ b/pixman/meson.build
@@ -30,7 +30,7 @@ version_h = configure_file(
   install_dir : get_option('prefix') / get_option('includedir') / 'pixman-1'
 )
 
-libpixman_extra_cargs = []
+libpixman_extra_cargs = get_option('cflags').split(',')
 default_library = get_option('default_library')
 if default_library != 'static' and cc.has_function_attribute('dllexport')
   libpixman_extra_cargs = ['-DPIXMAN_API=__declspec(dllexport)']
@@ -66,7 +66,7 @@ foreach simd : simds
     pixman_simd_libs += static_library(
       name,
       [name + '.c', config_h, version_h, simd[3]],
-      c_args : simd[2]
+      c_args : get_option('cflags').split(',') + simd[2], pic : get_option('b_staticpic')
     )
   endif
 endforeach
@@ -126,7 +126,7 @@ libpixman = library(
   dependencies : [dep_m, dep_threads],
   include_directories : cpu_features_inc,
   version : meson.project_version(),
-  install : true,
+  pic : get_option('b_staticpic'),
 )
 
 inc_pixman = include_directories('.')
