# Fixes linker errors when building for FreeBSD.

--- a/src/pulsecore/proplist-util.c
+++ b/src/pulsecore/proplist-util.c
@@ -111,7 +111,7 @@ void pa_init_proplist(pa_proplist *p) {
     const char *pp;
 
     pa_assert(p);
-
+#if 0
     if (environ) {
 
         /* Some applications seem to reset environ to NULL for various
@@ -147,7 +147,7 @@ void pa_init_proplist(pa_proplist *p) {
             }
         }
     }
-
+#endif
     if ((pp = getenv("PULSE_PROP"))) {
         pa_proplist *t;
 
