# Allows SDL to link statically against ANGLE.

--- a/meson.build
+++ b/meson.build
@@ -1171,13 +1171,13 @@ sdl2main_dep = declare_dependency(
 )
 
 deps = [core_deps, extra_deps]
-
+angle_dep = dependency('angle', required: get_option('angle'))
 sdl2 = library('sdl2',
     all_sources,
     include_directories : core_inc,
-    c_args : core_args,
+    c_args : core_args + (angle_dep.found() ? ['-DSDL_VIDEO_STATIC_ANGLE'] : []),
     link_args : core_ldflags,
-    dependencies : deps,
+    dependencies : deps + [angle_dep],
     override_options : ['c_std=none'],
     install : do_install,
     gnu_symbol_visibility: 'hidden',
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -238,3 +238,5 @@ option(
     deprecated: true,
     description : 'automatically link against sdl2main (deprecated, link to dependency sdl2_main instead)'
 )
+
+option('angle', type: 'feature', value: 'auto')
--- a/src/video/SDL_egl_c.h
+++ b/src/video/SDL_egl_c.h
@@ -23,6 +23,8 @@
 #ifndef SDL_egl_h_
 #define SDL_egl_h_
 
+extern int mkxp_angle_egl_is_wayland;
+
 #if SDL_VIDEO_OPENGL_EGL
 
 #include "SDL_egl.h"
--- a/src/video/wayland/SDL_waylandopengles.c
+++ b/src/video/wayland/SDL_waylandopengles.c
@@ -39,7 +39,9 @@ int Wayland_GLES_LoadLibrary(_THIS, const char *path)
 {
     int ret;
     SDL_VideoData *data = (SDL_VideoData *)_this->driverdata;
-
+#ifdef SDL_VIDEO_STATIC_ANGLE
+    mkxp_angle_egl_is_wayland = 1;
+#endif
     ret = SDL_EGL_LoadLibrary(_this, path, (NativeDisplayType) data->display, 0);
 
     Wayland_PumpEvents(_this);
--- a/src/video/x11/SDL_x11opengles.c
+++ b/src/video/x11/SDL_x11opengles.c
@@ -52,7 +52,9 @@ int X11_GLES_LoadLibrary(_THIS, const char *path)
         return SDL_SetError("SDL not configured with OpenGL/GLX support");
 #endif
     }
-
+#ifdef SDL_VIDEO_STATIC_ANGLE
+    mkxp_angle_egl_is_wayland = 0;
+#endif
     return SDL_EGL_LoadLibrary(_this, path, (NativeDisplayType) data->display, 0);
 }
 
