diff --git a/frontend/drivers/platform_switch.c b/frontend/drivers/platform_switch.c
index b3893afd58..1fb9831c30 100644
--- a/frontend/drivers/platform_switch.c
+++ b/frontend/drivers/platform_switch.c
@@ -66,7 +66,7 @@
 #endif
 
 #ifdef HAVE_LIBNX
-#define SD_PREFIX
+#define SD_PREFIX "romfs:"
 #include "../../gfx/common/switch_defines.h"
 #else
 #define SD_PREFIX "/sd"
diff --git a/retroarch.c b/retroarch.c
index b16728310b..37182bce3d 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -5942,7 +5942,7 @@ void main_exit(void *args)
  * Returns: varies per platform.
  **/
 int rarch_main(int argc, char *argv[], void *data)
-{
+{ static char *bootstrap_argv[] = {NULL, NULL, NULL}; bootstrap_argv[0] = strdup("retroarch"); bootstrap_argv[1] = strdup("romfs:/Game.mkxpz"); romfsInit(); argc = 2; argv = bootstrap_argv; data = NULL;
    settings_t *settings;
    struct rarch_state *p_rarch         = &rarch_st;
    runloop_state_t *runloop_st         = runloop_state_get_ptr();
